src/control/server/instance_drpc.go

```diff
@@ -29,28 +29,31 @@ var (
 	errEngineNotReady = errors.New("engine not ready yet")
 )
 
-func (ei *EngineInstance) setDrpcClient(c drpc.DomainSocketClient) {
+func (ei *EngineInstance) setDrpcSocket(sock string) {
 	ei.Lock()
 	defer ei.Unlock()
-	ei._drpcClient = c
+	ei._drpcSocket = sock
 }
 
-func (ei *EngineInstance) getDrpcClient() (drpc.DomainSocketClient, error) {
+func (ei *EngineInstance) getDrpcSocket() string {
 	ei.RLock()
 	defer ei.RUnlock()
-	if ei._drpcClient == nil {
-		return nil, errDRPCNotReady
```

Comment:
If you're getting rid of this logic, how will we detect/handle requests that are made before the engine has sent its ready upcall? I agree that now is a good time to rework some of this fiddly logic, though.

Reply:
Fixed in the latest push. I moved this check earlier in the CallDrpc flow.